### PR TITLE
Add upload to blob support and freeze example fairseq to 0.9

### PR DESCRIPTION
**Pull request by SwordFaith**
* Add blob model, controller, router in rest_server
* Add upload to blob logic in marketplace_api and generate_job_protocol
* Freeze example fairseq version to 0.9

### CODE REVIEW

File: rest_server/src/controllers/storage_controller.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    res.status(200).json(result);
+  } catch (e) {
+    databaseErrorHandler(e, res);
+  }
+});
+
+const create = asyncHandler(async (req, res, next) => {
+  try {
+    const id = await Blob.create(req.body);
+    console.log(id);
```
Comment (debuggy) [inline rest_server/src/controllers/storage_controller.js:25]: please delete all console.log functions

File: rest_server/src/models/index.js [hunk 2]
```diff
@@ -19,9 +20,21 @@ const sequelize = new Sequelize(DATABASE, DB_USERNAME, DB_PASSWORD, {
...
+// TODO: ensure connection succeeds
+sequelize
+  .authenticate()
+  .then(() => {
+    console.log('Connection has been established successfully.');
+  })
+  .catch(error => {
+    console.error('Unable to connect to the database:', error);
+  })
+  .done();
```
Comment (debuggy) [inline rest_server/src/models/index.js:32]: what does .done() do, is it necessary?

File: webportal/src/app/market_detail/utils/generate_job_protocol.js [hunk 3]
```diff
@@ -1,3 +1,30 @@
...
+    '\techo "Not found OUTPUT_DIR environ"',
+    'else',
+    '\tpai_copy upload ${OUTPUT_DIR} paiuploadtest ${PAI_USER_NAME}/${PAI_JOB_NAME}/', // eslint-disable-line no-template-curly-in-string
+    'fi',
+  ];
+  item.protocol.taskRoles.taskrole.commands = protocolHeaderArray.concat(
+    item.protocol.taskRoles.taskrole.commands,
+    protocolFooterArray,
+  );
+  console.log(item.protocol);
```
Comment (debuggy) [inline webportal/src/app/market_detail/utils/generate_job_protocol.js:28]: delete

File: webportal/src/app/utils/marketplace_api.js [hunk 4]
```diff
@@ -1,10 +1,42 @@
...
+  containerName,
+  type = 'blob',
+) {
+  const queryStr = queryString.stringify({
+    username,
+    storageAccount,
+    containerName,
+    type,
+  });
+  console.log(queryStr);
```
Comment (debuggy) [inline webportal/src/app/utils/marketplace_api.js:21]: delete

File: webportal/src/app/utils/marketplace_api.js [hunk 5]
```diff
@@ -1,10 +1,42 @@
...
+  const url = `${MARKETPLACE_API_URL}/storages/blobs?${queryStr}`;
+  const res = await fetch(url, {
+    method: 'GET',
+  });
+  if (res.ok) {
+    const blobs = await res.json();
+    if (isEmpty(blobs)) {
+      throw new Error('Error no available connectionString!');
+    }
+    console.log(blobs[0]);
```
Comment (debuggy) [inline webportal/src/app/utils/marketplace_api.js:31]: delete

File: rest_server/src/models/blob.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+      storageAccount,
+      containerName,
+      user,
+      this.models,
+    );
+  }
+
+  async create(blobReq) {
+    const handler = modelSyncHandler(async blobReq => {
+      const item = await this.orm.create(blobReq);
```
Comment (debuggy) [inline rest_server/src/models/blob.js:75]: needs validation

File: rest_server/src/models/blob.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+  async update(blobId, newBlobReq) {
+    const handler = modelSyncHandler(async (blobId, newBlobReq) => {
+      const blob = await this.orm.findOne({
+        where: { id: blobId },
+      });
+      if (isNil(blob)) {
+        return null;
+      } else {
+        await blob.update(newBlobReq);
```
Comment (debuggy) [inline rest_server/src/models/blob.js:105]: needs validation

File: rest_server/src/models/blob.js (hunk 6 above)
Comment (SwordFaith) [reply rest_server/src/models/blob.js:75]: Which kind of validation, I checked user.js and marketplace_item.js, is there need an authorized token ?

**Comment by debuggy** [comment]
LGTM :shipit:

**Review by debuggy — APPROVED** [review]

**Pull request merged**